operator/pkg/ciliumendpointslice/rate_limit.go

```diff
@@ -22,94 +23,73 @@ type rateLimit struct {
 
 type rateLimitConfig struct {
 	current          rateLimit
-	dynamicRateLimit []rateLimit
+	dynamicRateLimit dynamicRateLimit
 
 	rateLimiter *workqueue.BucketRateLimiter
 
 	logger logrus.FieldLogger
 }
 
+type dynamicRateLimit []rateLimit
+
+var defaultRateLimit = dynamicRateLimit{
+	{Nodes: 0, Limit: 10, Burst: 20},
+}
+
 func getRateLimitConfig(p params) rateLimitConfig {
-	var dynamicRateLimit []rateLimit
-	if p.Cfg.CESEnableDynamicRateLimit {
-		var err error
-		dynamicRateLimit, err = parseDynamicRateLimit(p.Cfg.CESDynamicRateLimitNodes, p.Cfg.CESDynamicRateLimitQPSLimit, p.Cfg.CESDynamicRateLimitQPSBurst)
+	dynamicRateLimit := defaultRateLimit
+	if len(p.Cfg.CESDynamicRateLimitConfig) != 0 {
+		parsed, err := parseDynamicRateLimit(p.Cfg.CESDynamicRateLimitConfig)
 		if err != nil {
-			p.Logger.WithError(err).Warn("Couldn't parse dynamic rate limit config")
+			p.Logger.WithError(err).Fatal("Couldn't parse CES rate limit config")
```

Comment:
Nit. AFAIR it is better to return an error to hive, rather than calling Fatal directly, to enable proper shutdown.

Reply:
Updated the surrounding functions to bubble the error up to hive